src/NuGet.Core/NuGet.DependencyResolver.Core/GraphModel/GraphOperations.cs

```diff
@@ -532,7 +539,7 @@ public static void ForEach<TItem>(this IEnumerable<GraphNode<TItem>> roots, Acti
             Cache<TItem>.ReleaseQueue(queue);
         }
 
-        public static void ForEach<TItem>(this GraphNode<TItem> root, Action<GraphNode<TItem>> visitor)
+        public static void ForEach<TItem>(this GraphNode<TItem> root, Action<GraphNode<TItem>> visitor, Func<GraphNode<TItem>, bool> skipNode = null)
```

Comment:
This is a public API change. Please create an overload.

Reply:
Updated.